commands/addResourceExtended.ts

```diff
@@ -0,0 +1,114 @@
+import { SlashCommandBuilder } from '@discordjs/builders';
+import { CommandInteraction, Message, MessageEmbed } from 'discord.js';
+import isValidUrl from '../utils/urlChecker';
+
+import { setCategorySelection, setBlockchainSelection, setLevelSelection, setMediaTypeSelection, setTagSelection } from './menuSelections';
+import { Resource } from '../types';
+
+export const data = new SlashCommandBuilder()
+    .setName('add-resource-extended')
+    .setDescription('Adds your link to the knowledgebase (must be airtable link)')
+    .addStringOption(
+        option => option.setRequired(true)
+        .setName('url')
+        .setDescription('Enter a link to a resource'));
+
+function ConfigureEmbed(embed: MessageEmbed, resource: Resource): MessageEmbed {
+  embed.setAuthor(resource.author ?? 'Author');
+  embed.setTitle(resource.title ?? 'Title');
+  embed.setURL(resource.source ?? 'Source');
+  embed.setDescription(resource.summary ?? 'Summary');
+  embed.setFields([
+    { name: 'level', value: resource.level ?? 'Level', inline: true },
+    { name: 'mediatype', value: resource.mediaType ?? 'Media Type', inline: true },
+    { name: 'blockchain', value: resource.blockchain ? resource.blockchain.join(', ') : 'Blockchain', inline: false },
+    { name: 'category', value: resource.category ? resource.category.join(', ') : 'Category', inline: true },
+    { name: 'tags', value: resource.tags ? resource.tags.join(', ') : 'Tags', inline: true },
+  ]);
+
+  return embed;
+}
+
+function UpdateEmbed(embed: MessageEmbed, embedMessage: Message, resource: Resource): void {
+  const updatedEmbed = ConfigureEmbed(new MessageEmbed(embed), resource);
+  embedMessage.edit({ embeds: [updatedEmbed] })
+}
+
+export async function execute(interaction: CommandInteraction) {
+    const userInput = interaction.options.getString('url')
+    if (userInput === undefined || userInput == null) {
+      return;
+    }
+
+    // TODO: Lookup the user's discord ID in AirTable and see if they are already a contributor
+    // TODO: Ask the user for their NFT ID if they have not contributed before
+
+    if (isValidUrl(userInput)) {
+        const newResource = new Resource();
+        newResource.source = userInput;
+        const filter = (m: Message) => interaction.user.id === m.author.id;
+        await interaction.reply('Please complete the addition process in the DM you just received');
```

Comment:
```suggestion
        await interaction.reply({ content: 'Please complete the addition process in the DM you just received', ephemeral: true });
```

This will make it so the bot doesn't clog the channel for others. the `ephermeral` tag make it so only the person interacting with the bot sees the messages from the bot.

Reply:
Yeah, not sure why I removed that.